wp-content/themes/pub/wporg-learn-2024/inc/block-config.php

```diff
@@ -29,9 +31,47 @@
  *
  * @return string The current URL.
  */
-function get_current_url() {
+function get_filtered_url() {
 	global $wp;
-	return home_url( add_query_arg( array(), $wp->request ) );
+
+	$current_url = home_url( add_query_arg( array(), $wp->request ) );
+
+	// Remove the page query var.
+	// If the path retains "page," the filtered result might be a 404 page.
+	$parsed_url = wp_parse_url( $current_url );
+	$path = isset( $parsed_url['path'] ) ? $parsed_url['path'] : '';
```

Comment:
I believe `$wp->request` will be just the path, no query args, so I don't think you need to do this extra parsing.

Reply:
bccee01